test/e2e/test/beat/builder.go

```diff
@@ -51,6 +51,12 @@ type Builder struct {
 }
 
 func (b Builder) SkipTest() bool {
+	// Beat on OpenShift3 requires specific securityContext due to hostPath volumes.
+	// Skipping all Beat tests to reduce maintenance burden.
+	if test.Ctx().Provider == "ocp3" {
+		return true
+	}
+
```

Comment:
What about Agent won't it run into the same issues?

Reply:
The agent tests in `agent/recipes_test.go` are already skipped (#4388) and the others work.